Feature/uh 004 update clients

sebastian0327 opened the pull request:
![image](https://github.com/user-attachments/assets/4bed9eef-9d9d-4fc7-a65a-65fd0c8bf9c7)
![image](https://github.com/user-attachments/assets/400a47b2-3fd7-49e4-aeb7-69cdf01137de)
![image](https://github.com/user-attachments/assets/f8d14b64-41c4-4e50-9667-cb875635f88f)

sebastian0327 committed:
feature(UH-004): update clients

sebastian0327 committed:
feature/UH-004-button-visual-effect

dgop92 commented on src/features/client/entities/client.ts:
```diff
@@ -0,0 +1,42 @@
+    phone: z.string()
+        .min(1, "Phone is required")
+        .max(255, "Phone is too long"),
+    corporateEmail: z.string()
+        .min(1, "Phone is required")
+        .max(255, "Phone is too long"),
```
email does not have it's corresponding validations

dgop92 commented on src/features/client/pages/ClientUpdate/useUpadteClient.ts:
```diff
@@ -0,0 +1,67 @@
+      queryClient.invalidateQueries({ queryKey: ["opportunities"] });
+
+      enqueueSnackbar(
+        "Client is updated successfully",
```
i think it is "was updated"

dgop92 reviewed (changes requested):
look at the comments

sebastian0327 committed:
fix/feature-UH-004/email-validation-&-syntax-error

sebastian0327 committed:
fix/feature/UH-004 put to patch

for better practices

sebastian0327 committed:
fix/UH-004-error-upadate-status

error that update all estatus in flase for default

sebastian0327 committed:
fix/feature/UH-004: button return

sebastian0327 committed:
fix/feature/UH-004-update-clients: error in patch

error for wrong paramts in useUpdateClient.ts
fix go back button link in ClientUpdatePage.tsx

dgop92 reviewed (approved):
ok

ndelarans reviewed (approved):
Ok

sebastian0327 committed:
fix/UH-004:padding

sebastian0327 committed:
fix/UH-004:padding

AlejandroSalas1143 reviewed (approved):
OK

dgop92 committed:
Merge branch 'main' into feature/UH-004-update-clients
